src/components/segmentedControl/index.tsx

```diff
@@ -170,39 +180,59 @@ const SegmentedControl = (props: SegmentedControlProps) => {
 
   const renderSegments = () =>
     _.map(segments, (_value, index) => {
+      const isLastSegment = index + 1 === segments?.length;
       return (
-        <Segment
-          key={index}
-          onLayout={onLayout}
-          index={index}
-          onPress={onSegmentPress}
-          selectedIndex={animatedSelectedIndex}
-          activeColor={activeColor}
-          inactiveColor={inactiveColor}
-          style={segmentsStyleProp}
-          segmentLabelStyle={segmentLabelStyle}
-          {...segments?.[index]}
-          testID={testID}
-        />
+        <>
+          <Segment
+            key={index}
+            onLayout={onLayout}
+            index={index}
+            onPress={onSegmentPress}
+            selectedIndex={animatedSelectedIndex}
+            activeColor={activeColor}
+            inactiveColor={inactiveColor}
+            style={[segmentsStyleProp]}
+            segmentLabelStyle={segmentLabelStyle}
+            iconTintColor={iconTintColor}
+            {...segments?.[index]}
+            testID={testID}
+          />
+          {!isLastSegment && (
+            <View
+              key={`segment.divider-${index}`}
+              width={segmentDividerWidth}
+              height={'100%'}
+              style={{backgroundColor: segmentDividerColor}}
+            />
+          )}
+        </>
       );
     });
-
   return (
     <View style={containerStyle} testID={testID}>
       <View row center style={[styles.container, style, {borderRadius, backgroundColor}]}>
+        <Reanimated.View
```

Comment:
Why separate into two Animated Views instead of one?

Reply:
When the outline and the active background are together the segment divider comes on top of the outline so the outline doesn't cover the divider. Moving the view to be on top of the segments make the active background be on top of the segments label therefore the label is not visible. `In short, moving it causes the outline to cover the divider like we want but the background also covers the segment label.`

Reply:
Let's talk tomorrow, I don't understand the issue...